src/App.tsx

```diff
@@ -1,24 +1,168 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { addTodo, getTodos, removeTodo } from './api/todos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { Todolist } from './components/Todolist';
+import { filterValues } from './constants';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+const USER_ID = 6438;
 
 export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [errorType, setErrorType] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [hasCompleted, setHasCompleted] = useState(false);
+  const [isTodoAdding, setIsTodoAdding] = useState(false);
+  const [isTodoRemoving, setIsTodoRemoving] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [selectedFilter, setSelectedFilter] = useState('All');
+  const [removingTodoIds, setRemovingTodoIds] = useState<number[]>([]);
+
+  const hasActive = todos.some(todoItem => !todoItem.completed);
+
+  const getTodosFromServer = async () => {
+    try {
+      const response = await getTodos(USER_ID);
+
+      setTodos(response);
+    } catch (error) {
+      setHasError(true);
+      setErrorType('upload');
+    }
+  };
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const filteredTodos = todos.filter(todo => {
+    if (selectedFilter === filterValues.completed) {
+      return todo.completed;
+    }
+
+    if (selectedFilter === filterValues.active) {
+      return !todo.completed;
+    }
+
+    return true;
+  });
+
+  const handleAddTodo = async () => {
+    try {
+      setIsTodoAdding(true);
+
+      const newTodo = {
+        userId: USER_ID,
+        title,
+        completed: false,
+      };
+
+      await addTodo(newTodo);
+
+      const demoTodo = {
+        ...newTodo,
+        id: 0,
+      };
+
+      setTempTodo(demoTodo);
+
+      await getTodosFromServer();
+
+      setHasError(false);
+    } catch (error) {
+      setHasError(true);
+      setErrorType('add');
+    } finally {
+      setIsTodoAdding(false);
+      setTempTodo(null);
+    }
+  };
+
+  // eslint-disable-next-line
```

Comment:
remove the disabling

Reply:
If I remove it, I get the linter error there. So I can't deploy this task to GH.
![Знімок екрана 2023-03-06 о 18 31 41](https://user-images.githubusercontent.com/112945779/223172164-ab7a3ef0-72a0-4f1e-af3b-e931a88ba8f7.png)

Reply:
You need fix eslint error and remove disabling lint. Try to google solution and fix error. [Maybe it helps](https://stackoverflow.com/questions/70702617/error-react-hook-usecallback-is-called-conditionally-react-hooks-must-be-cal), if not please ask questions in fe_chat and tag mentors. It will be faster, because your answers in GitHub will be visible to mentor only after you re-request review

Reply:
Fixed it, thanks!